src/slice.rs

```diff
@@ -143,35 +144,39 @@ impl<'a, T> NonEmptyIterator for NEChunks<'a, T> {
 
     fn next(&mut self) -> Option<Self::Item> {
         if self.index == 0 {
-            let end = self.window.get() - 1;
+            let end = min(self.window.get() - 1, self.tail.len());
```

Comment:
This is required to ensure cases where we have less elements in the collection than the given window.